jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowTypedWorkItemHandler.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.core;
+
+import org.kie.api.runtime.process.TypedWorkItem;
+import org.kie.api.runtime.process.TypedWorkItemHandler;
+
+public abstract class AbstractLogOrThrowTypedWorkItemHandler<W extends TypedWorkItem<?, ?>> extends AbstractLogOrThrowHandler implements TypedWorkItemHandler<W> {
```

Comment:
Sorry I have a question here maybe I don't understand something - in the doc for this it says "Seamlessly convert between TypedWorkItem<?,?> to WorkItem to retain backwards compatiblity ad interim." - on the contrary this seems somewhat invasive as workitems have to declare use of typed or not, not sure where the backwards compat is used - please explain.